### PR TITLE
Added Date Formatting Functionality

### LINKED ISSUE #7650: Add date formatting functionality
### Description

To avoid repetition, either add a date formatting helper function or add a date library such as date fns.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Vishrut19**
## Fixes Issue

Closes #7650

## Changes proposed
Added Date Formatting Functionality 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
Screenshot 2023-07-01 at 8 53 48 PM

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7909"></a>

### CODE REVIEW

File: components/event/EventCard.js [hunk 1]
```diff
@@ -57,12 +53,12 @@ export default function EventCard({ manage, event, usernames }) {
           {event.price?.startingFrom > 0 && <TbCoin title="Paid event" />}
           {event.price?.startingFrom === 0 && <TbCoinOff title="Free event" />}
         </div>
-        <div className="flex-1 space-y-1 p-4">
```
Comment (eddiejaoude) [inline components/event/EventCard.js:60]: I do not understand this change?

File: components/event/EventCard.js [hunk 2]
```diff
@@ -112,15 +108,15 @@ export default function EventCard({ manage, event, usernames }) {
...
-            <div className="isolate flex -space-x-1 ">
+            <div className="flex -space-x-1 isolate ">
               {usernames &&
                 usernames.map((username) => {
                   return (
                     <Link
                       href={`/${username}`}
                       key={username}
                       aria-label={`Visit user ${username}`}
-                      className="hidden lg:block h-10 w-10"
```
Comment (eddiejaoude) [inline components/event/EventCard.js:123]: this change looks the same?

File: components/event/EventCard.js [hunk 3]
```diff
@@ -19,20 +20,15 @@ export default function EventCard({ manage, event, usernames }) {
...
-      dateStyle: "full",
-      timeStyle: "long",
-    };
     try {
       setStartTime(
-        new Intl.DateTimeFormat("en-GB", dateTimeStyle).format(
-          new Date(event.date.start)
+        format(
+          new Date(event.date.start),
+          "eeee, d MMMM yyyy 'at' HH:mm:ss OOOO"
```
Comment (eddiejaoude) [inline components/event/EventCard.js:27]: I think the goal of the issue was to abstract the reuse of the date formatting similar to `services/utils/abbreviateNumbers.js`, I don't think need to add another library

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for contributing, I have added inline comments

**Comment by Vishrut19** [comment]
> Thank you for contributing, I have added inline comments

okay I will have a look and do the needful changes

File: components/event/EventCard.js (hunk 1 above)
Comment (Vishrut19) [reply components/event/EventCard.js:60]: This change is done by my prettier. So, it is just making tailwind css properties arrange alphabetically.

File: components/event/EventCard.js (hunk 2 above)
Comment (Vishrut19) [reply components/event/EventCard.js:123]: This change is done by my prettier. So, it is just making tailwind css properties arrange alphabetically.

File: components/event/EventCard.js (hunk 3 above)
Comment (Vishrut19) [reply components/event/EventCard.js:27]: got it as I was a bit confused as Emma had written in the issues section that either to write helper function or to add date-fns that's why. I will do the needful changes and try to raise new modified PR tomorrow.

File: components/event/EventCard.js (hunk 1 above)
Comment (eddiejaoude) [reply components/event/EventCard.js:60]: We don't have those Prettier settings in the project, please use the Project's settings and style

File: components/event/EventCard.js (hunk 3 above)
Comment (eddiejaoude) [reply components/event/EventCard.js:27]: I think it was an idea Emma mention, but not the goal. Ok sure, if you create a new PR please close this one, but it is best to add your changes to this PR if you can

File: components/event/EventCard.js (hunk 1 above)
Comment (Vishrut19) [reply components/event/EventCard.js:60]: How to find prettier settings that are present for the project?

File: components/event/EventCard.js (hunk 3 above)
Comment (Vishrut19) [reply components/event/EventCard.js:27]: So, where ever dates are present I need to change it with my helper function. Right?

File: components/event/EventCard.js (hunk 1 above)
Comment (eddiejaoude) [reply components/event/EventCard.js:60]: The settings are picked up automatically for me, maybe you have some settings overriding it

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution, the package lock file changes are not required

**Comment by eddiejaoude** [comment]
The package lock file has been deleted, this file is still required in the project, but your changes were not required.

Please remove changes to the lock file from your pull request (currently your changes are to delete the file)

**Comment by eddiejaoude** [comment]
Did you mean to close the PR?

**Comment by Vishrut19** [comment]
> Did you mean to close the PR?

not eddie but bymistake I deleted my previous commits.So because of which this branch also got marked as closed. Will now have to open a new PR and push the code

**Comment by eddiejaoude** [comment]
ok no problem, just checking everything was ok

**Comment by Vishrut19** [comment]
> ok no problem, just checking everything was ok

Created a new PR. Link to new PR - https://github.com/EddieHubCommunity/LinkFree/pull/7978#issuecomment-1620970982

**Pull request closed without merging**